File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/abstractfactory/app/Application.java
package com.ute.refactoring.abstractfactory.app;

import com.ute.refactoring.abstractfactory.buttons.Button;
import com.ute.refactoring.abstractfactory.checkboxes.Checkbox;
import com.ute.refactoring.abstractfactory.factories.GUIFactory;

public class Application {
    private final Button button;
    private final Checkbox checkbox;

    public Application(GUIFactory factory){
        button = factory.createButton();
        checkbox = factory.createCheckbox();
    }

    public  void paint(){
        button.paint();
        checkbox.paint();
    }
}

File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/AbstractFactory/App/Application.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.AbstractFactory
{
    public class Application
    {
        private IButton button;
        private Checkbox checkbox;

        public Application(GUIFactory factory)
        {
            button = factory.CreateButton();
            checkbox = factory.CreateCheckbox();
        }

        public void paint()
        {
            button.paint();
            checkbox.paint();
        }
    }
}
